BLOCKER modernizer-maven-plugin depends on plexus-utils but does not declare it as dependency

The `modernizer-maven-plugin` depends on plexus-utils (uses `org.codehaus.plexus.util.StringUtils` from it) **but does not declare it as dependency**. Maven 2.x did "auto-inject" plexus-utils into every Mojo, and Maven 3.x continued doing it for compatibility reasons, but this "backward compatilibility feature" will be removed in upcoming Maven 3.9.x and Maven 4.x. Hence, this plugin is broken when used with them.

Error it throws:
```
[ERROR] Failed to execute goal org.gaul:modernizer-maven-plugin:2.4.0:modernizer (modernizer) on project maven-indexer: Execution modernizer of goal org.gaul:modernizer-maven-plugin:2.4.0:modernizer failed: A required class was missing while executing org.gaul:modernizer-maven-plugin:2.4.0:modernizer: org/codehaus/plexus/util/StringUtils
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.gaul:modernizer-maven-plugin:2.4.0
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/home/cstamas/.m2/repository-oss/org/gaul/modernizer-maven-plugin/2.4.0/modernizer-maven-plugin-2.4.0.jar
[ERROR] urls[1] = file:/home/cstamas/.m2/repository-oss/backport-util-concurrent/backport-util-concurrent/3.1/backport-util-concurrent-3.1.jar
[ERROR] urls[2] = file:/home/cstamas/.m2/repository-oss/org/codehaus/plexus/plexus-interpolation/1.11/plexus-interpolation-1.11.jar
[ERROR] urls[3] = file:/home/cstamas/.m2/repository-oss/junit/junit/3.8.1/junit-3.8.1.jar
[ERROR] urls[4] = file:/home/cstamas/.m2/repository-oss/org/ow2/asm/asm/9.2/asm-9.2.jar
[ERROR] urls[5] = file:/home/cstamas/.m2/repository-oss/org/ow2/asm/asm-commons/9.2/asm-commons-9.2.jar
[ERROR] urls[6] = file:/home/cstamas/.m2/repository-oss/org/ow2/asm/asm-tree/9.2/asm-tree-9.2.jar
[ERROR] urls[7] = file:/home/cstamas/.m2/repository-oss/org/ow2/asm/asm-analysis/9.2/asm-analysis-9.2.jar
[ERROR] urls[8] = file:/home/cstamas/.m2/repository-oss/org/gaul/modernizer-maven-annotations/2.4.0/modernizer-maven-annotations-2.4.0.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] : org.codehaus.plexus.util.StringUtils
[ERROR] -> [Help 1]
```

To fix it:
* explicitly declare dependency on `org.codehaus.plexus:plexus-utils:3.5.0` (or version needed). This fix is backward but also forward compatible with upcoming Maven versions.

--- a/modernizer-maven-plugin/pom.xml
+++ b/modernizer-maven-plugin/pom.xml
@@ -17,12 +17,11 @@
   <description>Detect use of legacy APIs which modern Java versions supersede.</description>
 
   <prerequisites>
-    <maven>3.2.5</maven>
+    <maven>${mavenVersion}</maven>
   </prerequisites>
 
   <properties>
-    <groovy.version>3.0.2</groovy.version>
-    <maven.version>3.2.5</maven.version>
+    <mavenVersion>3.2.5</mavenVersion>
   </properties>
 
   <dependencies>
@@ -65,7 +64,7 @@
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
-      <version>4.13.1</version>
+      <version>4.13.2</version>
       <scope>test</scope>
     </dependency>
     <dependency>
@@ -157,7 +156,7 @@
           <dependency>
             <groupId>org.codehaus.groovy</groupId>
             <artifactId>groovy-all</artifactId>
-            <version>${groovy.version}</version>
+            <version>3.0.2</version>
             <type>pom</type>
           </dependency>
         </dependencies>

--- a/modernizer-maven-plugin/pom.xml
+++ b/modernizer-maven-plugin/pom.xml
@@ -17,11 +17,12 @@
   <description>Detect use of legacy APIs which modern Java versions supersede.</description>
 
   <prerequisites>
-    <maven>3.0.5</maven>
+    <maven>3.2.5</maven>
   </prerequisites>
 
   <properties>
     <groovy.version>3.0.2</groovy.version>
+    <maven.version>3.2.5</maven.version>
   </properties>
 
   <dependencies>
@@ -70,24 +71,20 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-plugin-api</artifactId>
-      <version>2.0</version>
+      <version>${maven.version}</version>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.plugin-tools</groupId>
       <artifactId>maven-plugin-annotations</artifactId>
-      <version>3.2</version>
+      <version>3.6.4</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
-      <artifactId>maven-project</artifactId>
-      <version>2.2.1</version>
-      <exclusions>
-        <exclusion>
-          <groupId>org.codehaus.plexus</groupId>
-          <artifactId>plexus-utils</artifactId>
-        </exclusion>
-      </exclusions>
+      <artifactId>maven-core</artifactId>
+      <version>${maven.version}</version>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.assertj</groupId>
@@ -113,6 +110,11 @@
       <groupId>org.gaul</groupId>
       <artifactId>modernizer-maven-annotations</artifactId>
       <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.codehaus.plexus</groupId>
+      <artifactId>plexus-utils</artifactId>
+      <version>3.5.0</version>
     </dependency>
   </dependencies>
 

--- a/modernizer-maven-plugin/pom.xml
+++ b/modernizer-maven-plugin/pom.xml
@@ -70,7 +70,7 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-plugin-api</artifactId>
-      <version>${maven.version}</version>
+      <version>${mavenVersion}</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
@@ -82,7 +82,7 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
-      <version>${maven.version}</version>
+      <version>${mavenVersion}</version>
       <scope>provided</scope>
     </dependency>
     <dependency>

